aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -431,18 +431,32 @@ static C10_ALWAYS_INLINE float32x4_t f32_fma_bf16(float32x4_t a, uint16x4_t b, u
   return f32_fma(a, to_bfloat16(b), to_bfloat16(c));
 }
 
+static C10_ALWAYS_INLINE float32x4_t f32_dot_bf16(float32x4_t a, bfloat16x8_t b, bfloat16x8_t c) {
+#ifdef __ARM_FEATURE_BF16
+  return vbfdotq_f32(a, b, c);
+#else
+  TORCH_CHECK(false, "unsupported operation used!");
```

Comment:
It should be a false static assert as this should be a compile fail, right? Can't imagine where you wanted this C++ function used by another call and not discover it until runtime.

Reply:
> Should this be a static_assert False situation?

It should be statically dead in this situation; the TORCH_CHECK is just a backstop in case that isn't true in the future by accident.